utils/slice.go

```diff
@@ -0,0 +1,18 @@
+// Copyright 2023 OWASP ModSecurity Core Rule Set Project
+// SPDX-License-Identifier: Apache-2.0
+
+package utils
+
+import (
+	"regexp"
+)
+
+func MatchSlice(r *regexp.Regexp, s []string) bool {
```

Comment:
We like to use meaningful names:
```suggestion
func MatchSlice(regex *regexp.Regexp, hayStack []string) bool {
```